src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -392,6 +392,14 @@ TransactionManager::submit_transaction_direct(
       journal->get_trimmer().update_journal_tails(
 	cache->get_oldest_dirty_from().value_or(start_seq),
 	cache->get_oldest_backref_dirty_from().value_or(start_seq));
+      if (is_trim_transaction(tref.get_src())) {
+	return journal->update_journal_tail(
```

Comment:
dirty_tail and alloc_tail are available to the journal directly.

Probably just:
```
seastar::future<> finish_commit(transaction_type_t type);
```
And move the condition into the circular journal.